packages/edit-site/src/components/global-styles/screen-block-list.js

```diff
@@ -45,6 +50,44 @@ function BlockMenuItem( { block } ) {
 }
 
 function ScreenBlockList() {
+	const [ filterValue, setFilterValue ] = useState( '' );
+	const debouncedSpeak = useDebounce( speak, 500 );
+	const { blockTypes, isMatchingSearchTerm } = useSelect( ( select ) => {
+		const { getBlockTypes } = select( blocksStore );
+		return {
+			blockTypes: getBlockTypes(),
+			isMatchingSearchTerm: select( blocksStore ).isMatchingSearchTerm,
+		};
+	}, [] );
+	const filteredBlockTypes = useMemo( () => {
+		if ( ! filterValue ) {
+			return blockTypes;
+		}
+		return blockTypes.filter( ( blockType ) =>
+			isMatchingSearchTerm( blockType, filterValue )
+		);
+	}, [ filterValue, blockTypes, isMatchingSearchTerm ] );
+
+	const blockTypesListRef = useRef();
+
+	// Announce search results on change
+	useEffect( () => {
+		if ( ! filterValue ) {
+			return;
+		}
+		// We extract the results from the wrapper div's `ref` because
+		// filtered items can contain items that will eventually not
+		// render and there is no reliable way to detect when a child
+		// will return `null`.
+		const count = blockTypesListRef.current.childElementCount;
```

Comment:
This is very weird 😬

Reply:
I can we still try to count using data (filter the results by checking if the block types actually have global styles panels, but not sure how complex would that be)

Reply:
>but not sure how complex would that be

I think it's very complex and involves many hooks with lots of logic and things like mapping props ([example](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/hooks.js#L171)), etc..

I agree it's very fragile but couldn't find a better solution for this without porting all the logic of hooks in selectors.

I added a comment about this to be easier to trace if this happens in the future..  Who knows, maybe React will support this info in a future version 😄

Reply:
I think it's fine to list blocks that don't have global styles properties in search results and if you select one it'd say "no style properties available" or something

Reply:
The number can differ a lot. For example without other plugins enabled the block types are `90`, but the ones with global style properties are `52`. While searching, the numbers will be smaller, but still it can result in many 'no need to be there' block types.